controllers/controller/workspacerouting/solvers/cluster_solver.go

```diff
@@ -74,17 +74,22 @@ func (s *ClusterSolver) GetExposedEndpoints(
 
 	for machineName, machineEndpoints := range endpoints {
 		for _, endpoint := range machineEndpoints {
-			if endpoint.Attributes[string(controllerv1alpha1.PUBLIC_ENDPOINT_ATTRIBUTE)] != "true" {
+			if endpoint.Exposure != devworkspace.PublicEndpointExposure {
 				continue
 			}
 			url, err := resolveServiceHostnameForEndpoint(endpoint, routingObj.Services)
 			if err != nil {
 				return nil, false, err
 			}
+			endpointAttributes := map[string]string{}
+			err = endpoint.Attributes.Into(&endpointAttributes)
```

Comment:
That's not really related to this PR:
That's going to be not nil if attributes contains any not string value, right? Like int, boolean and so on.
So, we probably need to validate the attributes earlier or skip/try to convert to string not string values.
But again, that's more related to issue about endpoints but not converting webhooks.

Reply:
I switch it to use `Attributes.Strings()` instead since the commits seem to be out of date with `devfile/api`. This should get around the issue of errant `bool` or `int`s getting in there.